lib/kafka_ex/server.ex

```diff
@@ -378,17 +396,27 @@ defmodule KafkaEx.Server do
 
       # credo:disable-for-next-line Credo.Check.Refactor.FunctionArity
       def retrieve_metadata(brokers, correlation_id, sync_timeout, topic \\ []), do: retrieve_metadata(brokers, correlation_id, sync_timeout, topic, @retry_count, 0)
+
+      # credo:disable-for-next-line Credo.Check.Refactor.FunctionArity
+      def retrieve_metadata_with_version(brokers, correlation_id, sync_timeout, api_version, topic \\ []), do: retrieve_metadata(brokers, correlation_id, sync_timeout, topic, @retry_count, api_version, 0)
```

Comment:
We should just add `api_version` as the last argument to `retrieve_metadata` and default it to `@default_api_version` (0). No need to add another function.

Reply:
done.